data_subscriber/daac_data_subscriber.py

```
@@ -331,101 +329,57 @@ def update_granule_index(es_spatial_conn, granule):
-    username, _, password = netrc.netrc().authenticators(edl)
+        if expiration_date <= now:
+            _delete_token(edl, username, password, token_dict['access_token'])
 
-    list_response = requests.get(token_list_url, auth=HTTPBasicAuth(username, password))
-    list_content = json.loads(list_response.content)
 
-    if not list_content:
-        create_response = requests.post(token_create_url, auth=HTTPBasicAuth(username, password))
-        response_content = json.loads(create_response.content)
+def _create_token(edl: str, username: str, password: str):
+    token_create_url = f"https://{edl}/api/users/token"
+
+    create_response = requests.post(token_create_url, auth=HTTPBasicAuth(username, password))
+    response_content = json.loads(create_response.content)
 
-        if "error" in response_content.keys():
-            logging.warning("Failed to acquire CMR token")
-            raise Exception(response_content['error'])
+    if "error" in response_content.keys():
+        logging.warning("Failed to acquire CMR token")
+        raise Exception(response_content['error'])
 
-        token = response_content["access_token"]
-    else:
-        token = list_content[0]["access_token"]
+    token = response_content["access_token"]
 
-    return {"token": token, "username": username, "password": password}
+    return token
 
 
-def _delete_token(edl: str, token_dict: dict) -> None:
+def _delete_token(edl: str, username: str, password: str, token: str) -> None:
     url = f"https://{edl}/api/users/revoke_token"
     try:
-        resp = requests.post(url, auth=HTTPBasicAuth(token_dict['username'], token_dict['password']),
-                             params={'token': token_dict['token']})
+        resp = requests.post(url, auth=HTTPBasicAuth(username, password),
+                             params={'token': token})
         if resp.status_code == 200:
             logging.info("CMR token successfully deleted")
         else:
```

consider writing the response here, or elaborating the the type of failure. Because there is a difference between a client error (deleting an expired token) and an internal server error (500). The current implementation would swallow any CMR server error, making it harder to troubleshoot.

---

```python
            logging.warning(f"CMR token deletion failed: {resp.status_code=}")
```